integration-tests/bats/stats.bats

```diff
@@ -116,6 +116,52 @@ teardown() {
     [ "${lines[1]}" = "0" ]
 }
 
+@test "stats: correct stats directory location, issue#8324" {
+    cd repo2
+
+    dolt sql -q "insert into xy values (0,0), (1,1)"
+
+    dolt sql -q "call dolt_stats_restart()"
+
+    run dolt sql -r csv -q "select count(*) from dolt_statistics"
+    [ "$status" -eq 0 ]
+
+    run stat .dolt/repo2
```

Comment:
Test is a little wonky. Probably should look for the file you want, not the one you don't.

Reply:
yeah, unfortunately we make the files on `dolt init`. i added a second test that verifies that a round-trip picks up the expected data